bundles/org.openhab.binding.modbus.e3dc/src/main/java/org/openhab/binding/modbus/e3dc/internal/handler/E3DCThingHandler.java

```diff
@@ -0,0 +1,453 @@
+    static final String DATA_READ_ERROR = "Data Modbus Read Error";
+
+    static final String INFO_GROUP = "info";
+    static final String EMERGENCY_GROUP = "emergency";
+    static final String POWER_GROUP = "power";
+    static final String STRINGS_GROUP = "strings";
+
+    private ChannelUID modbusIdChannel;
+    private ChannelUID modbusVersionChannel;
+    private ChannelUID supportedRegistersChannel;
+    private ChannelUID manufacturerChannel;
+    private ChannelUID modelNameChannel;
+    private ChannelUID serialNumberChannel;
+    private ChannelUID firmwareChannel;
+
+    private ChannelUID epStatusChannel;
+    private ChannelUID batteryChargingLockedChannel;
+    private ChannelUID batteryDischargingLockedChannel;
+    private ChannelUID epPossibleChannel;
+    private ChannelUID weatherPredictedChargingChannel;
+    private ChannelUID regulationStatusChannel;
+    private ChannelUID chargeLockTimeChannel;
+    private ChannelUID dischargeLockTimeChannel;
+
+    private ChannelUID pvPowerSupplyChannel;
+    private ChannelUID batteryPowerSupplyChannel;
+    private ChannelUID batteryPowerConsumptionChannel;
+    private ChannelUID householdPowerConsumptionChannel;
+    private ChannelUID gridPowerConsumpitionChannel;
+    private ChannelUID gridPowerSupplyChannel;
+    private ChannelUID externalPowerSupplyChannel;
+    private ChannelUID wallboxPowerConsumptionChannel;
+    private ChannelUID wallboxPVPowerConsumptionChannel;
+    private ChannelUID autarkyChannel;
+    private ChannelUID selfConsumptionChannel;
+    private ChannelUID batterySOCChannel;
+
+    private ChannelUID string1AmpereChannel;
+    private ChannelUID string1VoltChannel;
+    private ChannelUID string1WattChannel;
+    private ChannelUID string2AmpereChannel;
+    private ChannelUID string2VoltChannel;
+    private ChannelUID string2WattChannel;
+    private ChannelUID string3AmpereChannel;
+    private ChannelUID string3VoltChannel;
+    private ChannelUID string3WattChannel;
+
+    private final ArrayList<E3DCWallboxThingHandler> listeners = new ArrayList<E3DCWallboxThingHandler>();
+    private final Logger logger = LoggerFactory.getLogger(E3DCThingHandler.class);
+    private final Parser dataParser = new Parser(DataType.DATA);
+    private ReadStatus dataRead = ReadStatus.NOT_RECEIVED;
+    private final Parser infoParser = new Parser(DataType.INFO);
+    private ReadStatus infoRead = ReadStatus.NOT_RECEIVED;
+    private @Nullable PollTask infoPoller;
+    private @Nullable PollTask dataPoller;
+    private @Nullable E3DCConfiguration config;
+
+    /**
+     * Communication interface to the slave endpoint we're connecting to
+     */
+    protected volatile @Nullable ModbusCommunicationInterface comms = null;
+    private int slaveId;
+
+    public E3DCThingHandler(Bridge thing) {
+        super(thing);
+
+        modbusIdChannel = channelUID(thing, INFO_GROUP, MODBUS_ID_CHANNEL);
```

Comment:
`E3DCBindingConstants` would be a better place for these. You could also make them static final.

Reply:
Are you really sure? Channel is bound to  thing.getUID. I placed it into the constructor because for each thing an own Channel needs to be allocated.  So I don't think static final is fitting for updating different Things.

Reply:
You're right, they can't be made static.